Sources/OTel/Metrics/OTelInstrument/Counter+OTelInstrument.swift

```diff
@@ -0,0 +1,54 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2024 Moritz Lang and the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import Tracing
+
+extension Counter: OTelMetricInstrument {
+    /// Return the current state as an OTel metric data point.
+    ///
+    /// Since our simplifed Swift Metrics backend datamodel only stores the current count, the only sensible mapping to
+    /// an OTel data point we can provide is a sum, with cumulative aggregation temporality.
+    func measure() -> OTelMetricPoint {
+        measure(instant: DefaultTracerClock.now)
+    }
+
+    /// Return the current state as an OTel metric data point.
+    ///
+    /// Since our simplifed Swift Metrics backend datamodel only stores the current count, the only sensible mapping to
+    /// an OTel data point we can provide is a sum, with cumulative aggregation temporality.
+    func measure(instant: some TracerInstant) -> OTelMetricPoint {
+        let doubleValue = Double(bitPattern: floatAtomic.load(ordering: .relaxed))
+        let intValue = intAtomic.load(ordering: .relaxed)
+        let pointValue: OTelNumberDataPoint.OTelPointValue
+        if !doubleValue.isZero {
+            pointValue = .double(doubleValue + Double(intValue))
+        } else {
+            pointValue = .int64(intValue)
+        }
+        return OTelMetricPoint(
+            name: name,
+            description: "",
```

Comment:
Should we provide default values in the `OTelMetricPoint` initializer instead?

Reply:
I've left this one as-is, for now, because:

1. We have an open question on the covering issue on how to deal with description and unit.
2. I expect that these types will be demoted to `package` access, as called out in the PR description.

The API/SDK spec dictates that they should be optional at the point the user interacts with the API, but I think this layer should map to the proto 1:1, where these are not optional. (Yes, they are using the optional support in proto3, and they document each field that is optional too in the comments in the proto).